### PR TITLE
Feature decorate class

### PR DESCRIPTION
**Pull request by biruk-bereka**
## In this PR I:
   - Created a nameable interface 
   - Created Base Decorator which extends nameable 
   - Created CapitalizeDecorator and TrimmerDecorator which extends Base Decorator 
  - Updated Person class inheriting from nameable

### CODE REVIEW

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @biruk-bereka, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Code is organized and clean
- You implemented the `Decorator` and `Namable` classes as required. Nice work.

## Required Changes ♻️
_Check the comments under the review._
- Please implement the following decorators:
```ruby
# Inherits from Decorator
class TrimmerDecorator < Decorator
    # The name has a maximum of 10 characters. If it's longer it should trim the word.
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
end
```
```ruby
# Inherits from Decorator
class CapitalizeDecorator < Decorator
    # Capitalize the first letter of the given name.
    def correct_name
      @nameable.correct_name.capitalize
    end
end
```
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_na
```
Comment (OyePriscilla) [inline trimmer_decorator.rb]: - [ ] Please kindly ensure to implement this variable name correctly, it's introducing bug to your app at the moment if I input a name `< 10.`

Untitled

```suggestion
    @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
```

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @biruk-bereka 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Use the Decorator design pattern ✔️
Decorator class inherits from Namable ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @biruk-bereka, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good  implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**